shapely/ops.py

```diff
@@ -179,6 +179,63 @@ def triangulate(geom, tolerance=0.0, edges=False):
     gc = geom_factory(func(geom._geom, tolerance, int(edges)))
     return [g for g in gc.geoms]
 
+
+def voronoi_diagram(geom, envelope=None, tolerance=0.0, edges=False):
+    """
+    Constructs a Voronoi Diagram [1] from the given geometry.
```

Comment:
Most of this is over my head but it looks really nice. I think you need a underscore after the references in numpydocs if we want it to build with hyperlinks eventually. Like this: `[1]_`. And you'll also want to add a `.. ` before each reference down in the References section.  

See point number 14 here: https://numpydoc.readthedocs.io/en/latest/format.html

Reply:
I'll defer to @sgillies on this, I'm happy to add the extra formatting if the docstrings will be built into documentation, but if they're mostly used for inline help (like `help(voronoi_diagram)`) then the extra formatting might not be helpful. Either way thanks for the suggestion! =)